### DISCUSSION TITLE
how to zoom in/out in a canvas

### DISCUSSION THREAD

**Discussion by GitSleep8** [seleniumbase/SeleniumBase#2054] [General]
Hi Michael, I want to move the mouse to a canvas and zoom in/out in the canvas, but it seems that Actionchains doesn't have this operation. Is there a way to simulate the scroll wheel to zoom in/out by Seleniumbase?

**Comment by mdmintz** [comment]
Maybe this:

```python
from seleniumbase import BaseCase
BaseCase.main(__name__, __file__)

class CanvasTests(BaseCase):
    def test_canvas(self):
        self.open("https://seleniumbase.io/canvas/")
        zoom_in = "canvas{zoom: 1.4;-moz-transform: scale(1.4);}"
        self.add_css_style(zoom_in)
        self.sleep(2)
```

**Reply by GitSleep8** [reply]
Thanks, it changes the style, but it can't trigger the event of the canvas. The canvas will show more points when zoom in and hide some points when zoom out for performance reason.  Sometimes zoom in/out will get stuck when there are millions of points, so I need the automatic test case to cover it, but I can't find a way to zoom in/out like we manually scroll the mouse wheel.

**Reply by mdmintz** [reply]
This is probably what you're looking for: https://stackoverflow.com/a/47287595/7058266

**Reply by GitSleep8** [reply]
Thanks a lot, it works!!
